cardano-node/src/Cardano/CLI/Tx/BenchmarkingNodeToNode.hs

```diff
@@ -0,0 +1,109 @@
+{-# LANGUAGE DataKinds           #-}
+{-# LANGUAGE GADTs               #-}
+{-# LANGUAGE RankNTypes          #-}
+{-# LANGUAGE ScopedTypeVariables #-}
+{-# LANGUAGE TupleSections       #-}
+{-# LANGUAGE TypeApplications    #-}
+
+module Cardano.CLI.Tx.BenchmarkingNodeToNode
+  ( BenchmarkTxSubmitTracers (..)
+  , SendRecvConnect
+  , SendRecvTxSubmission
+  , benchmarkConnectTxSubmit
+  )
+where
+
+import           Prelude
+import qualified Control.Concurrent as Concur
+import           Cardano.Prelude (Void, forever)
+
+import qualified Codec.CBOR.Term as CBOR
+import           Codec.Serialise (DeserialiseFailure)
+import           Control.Monad.Class.MonadTimer (MonadTimer, threadDelay)
+import           Control.Tracer (Tracer, nullTracer)
+import           Data.ByteString.Lazy (ByteString)
+import           Network.Mux.Interface (AppType(InitiatorApp))
+import           Network.Socket (AddrInfo, SockAddr)
+import           Network.TypedProtocol.Driver (TraceSendRecv, runPeer)
+import           Network.TypedProtocol.Driver.ByteLimit (DecoderFailureOrTooMuchInput)
+import           Ouroboros.Consensus.Block (BlockProtocol)
+import           Ouroboros.Consensus.Mempool.API (GenTxId, GenTx)
+import           Ouroboros.Consensus.Node.Run (RunNode)
+import           Ouroboros.Consensus.NodeNetwork (ProtocolCodecs(..), protocolCodecs)
+import           Ouroboros.Consensus.Protocol.Abstract (NodeConfig)
+import           Ouroboros.Network.Mux (OuroborosApplication(..))
+import           Ouroboros.Network.NodeToNode
```

Comment:
We avoid use of implicit imports -- this should be wither qualified (`as Network`, for example) or explicit.